tools/state-viewer/src/apply_chain_range.rs

```diff
@@ -353,30 +353,33 @@ pub fn apply_chain_range(
         ?storage)
     .entered();
     let chain_store = ChainStore::new(read_store.clone(), genesis.config.genesis_height, false);
+    let final_head = chain_store.final_head().unwrap();
+    let shard_layout = epoch_manager.get_shard_layout(&final_head.epoch_id).unwrap();
+    let shard_uid =
+        near_primitives::shard_layout::ShardUId::from_shard_id_and_layout(shard_id, &shard_layout);
     let (start_height, end_height) = match mode {
         ApplyRangeMode::Benchmarking => {
```

Comment:
Q: shouldn't we load flat and memtries in whatever mode has been requested too, not just for benchmarking?

Reply:
Another observation: if we want `StorageSource` arguments to work consistently wherever they are available, it would seem to me that we might want `--load-flat`/`--load-memtries` instead of whatever we have right now and have these boolean like arguments act as sort-of a unit type that implements logic for loading said storage types.

The current mechanism is kinda awkward.